pymongo/client_options.py

```diff
@@ -69,43 +69,51 @@ def _parse_read_concern(options):
 
 def _parse_ssl_options(options):
     """Parse ssl options."""
-    use_ssl = options.get('ssl')
-    if use_ssl is not None:
-        validate_boolean('ssl', use_ssl)
-
-    certfile = options.get('ssl_certfile')
-    keyfile = options.get('ssl_keyfile')
-    passphrase = options.get('ssl_pem_passphrase')
-    ca_certs = options.get('ssl_ca_certs')
-    cert_reqs = options.get('ssl_cert_reqs')
-    match_hostname = options.get('ssl_match_hostname', True)
-    crlfile = options.get('ssl_crlfile')
-    check_ocsp_endpoint = options.get('ssl_check_ocsp_endpoint', True)
-
-    ssl_kwarg_keys = [k for k in options
-                      if k.startswith('ssl_') and options[k]]
-    if use_ssl is False and ssl_kwarg_keys:
-        raise ConfigurationError("ssl has not been enabled but the "
-                                 "following ssl parameters have been set: "
-                                 "%s. Please set `ssl=True` or remove."
-                                 % ', '.join(ssl_kwarg_keys))
-
-    if ssl_kwarg_keys and use_ssl is None:
-        # ssl options imply ssl = True
-        use_ssl = True
-
-    if use_ssl is True:
+    use_tls = options.get('tls')
+    if use_tls is not None:
+        validate_boolean('tls', use_tls)
+
+    certfile = options.get('tlscertificatekeyfile')
+    passphrase = options.get('tlscertificatekeyfilepassword')
+    ca_certs = options.get('tlscafile')
+    allow_invalid_certificates = options.get('tlsallowinvalidcertificates')
+    allow_invalid_hostnames = options.get('tlsallowinvalidhostnames', False)
+    crlfile = options.get('tlscrlfile')
+    disable_ocsp_endpoint_check = options.get('tlsdisableocspendpointcheck', False)
+
+    enabled_tls_opts = []
+    for opt in ('tlscertificatekeyfile', 'tlscertificatekeyfilepassword',
+                'tlscafile', 'tlscrlfile'):
+        # Any non-null value of these options implies tls=True.
+        if opt in options and options[opt]:
+            enabled_tls_opts.append(opt)
+    for opt in ('tlsallowinvalidcertificates', 'tlsallowinvalidhostnames',
+                'tlsdisableocspendpointcheck'):
+        # A value of False for these options implies tls=True.
+        if opt in options and options[opt] is False:
+            enabled_tls_opts.append(opt)
+
+    if enabled_tls_opts:
+        if use_tls is False:
```

Comment:
Suggest switching the order of the this `if` so we can replace the `use_tls is False` with `not use_tls`.